cmd/external-repos/main.go

```diff
@@ -290,6 +297,86 @@ func enqueueSnapshotRepos(ctx context.Context, urls *[]string, interval *int) er
 	return nil
 }
 
+func enqueueSnapshotsCleanup(ctx context.Context, olderThanDays int) error {
+	q, err := queue.NewPgQueue(ctx, db.GetUrl())
+	if err != nil {
+		return fmt.Errorf("error getting new task queue: %w", err)
+	}
+	c := client.NewTaskClient(&q)
+	repoConfigDao := dao.GetRepositoryConfigDao(db.DB, pulp_client.GetPulpClientWithDomain(""))
+	snapshotDao := dao.GetSnapshotDao(db.DB)
+	taskInfoDao := dao.GetTaskInfoDao(db.DB)
+
+	repoConfigs, err := repoConfigDao.ListReposWithOutdatedSnapshots(ctx, olderThanDays)
+	if err != nil {
+		return fmt.Errorf("error getting repository configurations: %v", err)
+	}
+
+	for _, repo := range repoConfigs {
+		// Fetch snapshots for repo and find those which are to be deleted
+		snaps, err := snapshotDao.FetchForRepoConfigUUID(ctx, repo.UUID)
+		if err != nil {
+			return fmt.Errorf("error fetching snapshots for repository %v", repo.Name)
+		}
+		if len(snaps) < 2 {
+			return nil
```

Comment:
i think this will stop checking repos once it gets to the first repo with just 1 snapshot. you might want to `continue` here instead of returning?